### ISSUE TITLE
Increase block count limit 32768

### ISSUE THREAD

**Issue by gitownit** [Parchive/par2cmdline#178]
This 32768 block limit make the block size a little giant when small random error recovery is the main goal, like for 512 byte/sector of a little defective hard drive (without AF).

Example :
One 200MB file result in about 32768 block count of 6000 byte size per block, while I would expect 1024 byte per block. So my recovery block possibilities are divided by six... and I let you imagine the recovery block lose with an 2GB filesize. 
There it take about 7 seconds to create this PAR with about 10 recovery block, and I largely can wait ten more time to create a better efficient PAR file. (on an old core2 arch at about 3.6Ghz)

So, PLEASE! I suggest you to increase this tiny limit, that is not needed anymore twenty years later, while we are now working with files of gigas.

I have give a look at the source and spotted some conditions about this 32768, but I'm asking me if there are not other stuff under the hood to take into account.
I think having a 24-bit limit (or even 32-bit) for the block count is the change todo.

Thanks!

**Comment by animetosho** [comment]
PAR2 is spec'd to use 16-bit GF, hence the file format is fundamentally restricted to 32768 input blocks.  
To go beyond that, you need a different file format.

Note that the block size doesn't have to match some underlying sector size - they can be larger and still work fine.

There's a [draft PAR3 specification](https://github.com/Parchive/par3cmdline) which currently supports arbitrary sized GF, which enables more than 32768 input blocks.

**Comment by miaLDN** [comment]
Is this where the extremely tiny max file limit comes from?

@BlackIkeEagle you mentioned this on #208 as completed for Version 1.0.0 ?

I'm close to a PiB of main storage now; and my file count is many many orders of magnitude above this ancient 16-bit limit.

```txt
With PAR 2.0 you can create PAR2 recovery files for as few as 1 or as many as 32768 files
```

**Comment by BlackIkeEagle** [comment]
@miaLDN yes as completed as a support ticket, where this is not something we can change in this project, see https://github.com/Parchive/par2cmdline/issues/178#issuecomment-1471002858
